### PR TITLE
Improvements: Allowed actions transactionally

### PR DESCRIPTION
**Pull request by lciolecki**
Add transactionally on mark process as achived or unarchived.

### CODE REVIEW

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessService.scala [hunk 1]
```diff
@@ -72,41 +78,31 @@ class ProcessService(managerActor: ActorRef,
...
-      archive <- writeRepository.archive(processId = process.idWithName.id, isArchived = false)
-      _ <- processActionRepository.markProcessAsUnArchived(process.idWithName.id, process.processVersionId, None)
-    } yield archive
-
-   def deployProcess(processIdWithName: ProcessIdWithName, savepointPath: Option[String], comment: Option[String])(implicit ec: ExecutionContext, user: LoggedUser): Future[EmptyResponse] =
+  def deployProcess(processIdWithName: ProcessIdWithName, savepointPath: Option[String], comment: Option[String])(implicit ec: ExecutionContext, user: LoggedUser): Future[EmptyResponse] =
      doAction(ProcessActionType.Deploy, processIdWithName, savepointPath, comment){ (processIdWithName: ProcessIdWithName, savepointPath: Option[String], comment: Option[String]) =>
        (managerActor ? Deploy(processIdWithName, user, savepointPath, comment))
-         .map(_ => Right(Unit))
+         .map(_ => ().asRight)
```
Comment (mproch) [inline ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessService.scala:99]: why is it better?

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/WriteProcessRepository.scala [hunk 2]
```diff
@@ -24,40 +23,38 @@ import pl.touk.nussknacker.ui.process.repository.WriteProcessRepository.UpdatePr
...
 
-    new DbWriteProcessRepository[Future](dbConfig, modelData.mapValues(_.migrations.version)) with WriteProcessRepository with BasicRepository
+    new DbWriteProcessRepository[Future](dbConfig, modelData.mapValues(_.migrations.version)) with WriteProcessRepository[DB] with BasicRepository
 
   case class UpdateProcessAction(id: ProcessId, deploymentData: ProcessDeploymentData, comment: String)
 
 }
 
-trait WriteProcessRepository {
+trait WriteProcessRepository[F[_]] {
```
Comment (mproch) [inline ui/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/WriteProcessRepository.scala:41]: Let's add comment it will be refactored/split

**Review by mproch — COMMENTED** [review]
:+1: please, just add comment about future work :)

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessService.scala (hunk 1 above)
Comment (lciolecki) [reply ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessService.scala:99]: Just look better :)

**Review by mproch — APPROVED** [review]

**Pull request merged**